src/ucp/core/ucp_ep.c

```diff
@@ -551,6 +551,35 @@ void ucp_worker_destroy_mem_type_endpoints(ucp_worker_h worker)
     UCS_ASYNC_UNBLOCK(&worker->async);
 }
 
+void ucp_worker_mem_type_endpoints_print_info(ucp_worker_h worker, FILE *stream)
+{
+    ucs_memory_type_t mem_type;
+    ucp_ep_h ep;
+
+    ucs_memory_type_for_each(mem_type) {
+        ep = worker->mem_type_ep[mem_type];
+        if (ep == NULL) {
+            continue;
+        }
+
+        ucp_ep_print_info(ep, stream);
+    }
+}
+
+static ucs_memory_type_t
+ucp_worker_lookup_ep_mem_type(ucp_worker_h worker, ucp_ep_h ep)
+{
+    ucs_memory_type_t mem_type;
+
+    ucs_memory_type_for_each(mem_type) {
+        if (ep == worker->mem_type_ep[mem_type]) {
+            return mem_type;
+        }
+    }
+
+    return UCS_MEMORY_TYPE_LAST;
```

Comment:
seems a bit ugly for endpoint to look up itself.
maybe add ucp_ep_print_info_internal() which will also accept a string of custom title for the endpoint. and in our case it will be the memory type name

Reply:
done